### ISSUE TITLE
[REVIEW]: PyVista: 3D plotting and mesh analysis through a streamlined interface for the Visualization Toolkit (VTK)

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1450]
**Submitting author:** @banesullivan (<a href="http://orcid.org/0000-0001-8628-4566">C. Bane Sullivan</a>)
**Repository:** <a href="https://github.com/pyvista/pyvista/" target ="_blank">https://github.com/pyvista/pyvista</a>
**Version:** 0.20.2
**Editor:** @leouieda
**Reviewer:** @Chilipp, @nicoguaro
**Archive:** <a href="https://doi.org/10.5281/zenodo.2647611" target="_blank">10.5281/zenodo.2647611</a>

## Status

[![status](http://joss.theoj.org/papers/78f2901bbdfbd2a6070ec41e8282d978/status.svg)](http://joss.theoj.org/papers/78f2901bbdfbd2a6070ec41e8282d978)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/78f2901bbdfbd2a6070ec41e8282d978"><img src="http://joss.theoj.org/papers/78f2901bbdfbd2a6070ec41e8282d978/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/78f2901bbdfbd2a6070ec41e8282d978/status.svg)](http://joss.theoj.org/papers/78f2901bbdfbd2a6070ec41e8282d978)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/vtkiorg/vtki" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@Chilipp & @nicoguaro, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @leouieda know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/vtkiorg/vtki">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 0.20.2
- [x] **Authorship:** Has the submitting author (@banesullivan) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/vtkiorg/vtki">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 0.20.2
- [x] **Authorship:** Has the submitting author (@banesullivan) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @Chilipp, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by banesullivan** [comment]
Probably not a big issue but I see it's on the checklist - `@whedon` marks version `0.18.1` in the checklist but PyVista is released officially on `0.20.1`

**Comment by nicoguaro** [comment]
I did a first review of the package and I have the following comments:

- The installation instructions for ``pip`` work properly while they do not work for ``conda``.
- The examples work properly, although the colors differ from the documentation.

**Comment by banesullivan** [comment]
Thanks for the comments @nicoguaro 

> The installation instructions for pip work properly while they do not work for conda.

we are currently waiting on the Conda-forge team to merge our anaconda recipe in https://github.com/conda-forge/staged-recipes/pull/8394

> The examples work properly, although the colors differ from the documentation.

This has to do with how we allow users to set plotting themes via `pyvista.set_plot_theme` the default theme a user experiences is a bit different than the `'document'` theme we use when building the docs. To have the same colors, you could add:

```py
import pyvista as pv
pv.set_plot_theme('document')
```

which is what we add in out `conf.py` when building the documentation/examples

**Comment by banesullivan** [comment]
Live on Conda-forge: [![conda-forge/pyvista](https://img.shields.io/conda/vn/conda-forge/pyvista.svg)](https://anaconda.org/conda-forge/pyvista)

**Comment by Chilipp** [comment]
Dear @banesullivan,

I am done with the review and I have to say that this is a very nice package that provides valuable new functionalities for visualizations with python. I worked with Paraview before and it's Python interface, and I have to say that your methodologies are both, useful and well-documented.

There are only a few minor issues:

- [ ] **version:** The version is not set correctly (as already mentioned in https://github.com/openjournals/joss-reviews/issues/1450#issuecomment-492497996 by @banesullivan)
- [ ] **dependendies:** You list the optional dependencies in the installation instructions, but not the necessary dependencies. You should be more transparent here
- [ ] **tests:** Although you state in your [CONTRIBUTING.md](https://github.com/pyvista/pyvista/blob/master/CONTRIBUTING.md#testing) how to run the tests locally, you do not state this in the installation instructions. I recommend to change this. You should at least include a link to the CONTRIBUTING.md file, such that someone who wants to verify the installation before using it, knows how to do this.

Otherwise, if the three issues are resolved, I recommend to accept this publication.

**Comment by banesullivan** [comment]
@Chilipp, Please see changes in https://github.com/pyvista/pyvista/commit/6e0762b7ae946b6141746ba134d0b3526ee356f5 and https://github.com/pyvista/pyvista/commit/9563643928a2f907e3001c481194adccacf62c48 addressing your comment.

## Dependencies

> You list the optional dependencies in the installation instructions, but not the necessary dependencies. You should be more transparent here

I have updated the installation instructions to include an itemized list of PyVista's required dependencies and a brief description for why - please note the list here:  https://docs.pyvista.org/getting-started/installation.html#dependencies


## Testing

> Although you state in your CONTRIBUTING.md how to run the tests locally, you do not state this in the installation instructions. I recommend to change this. You should at least include a link to the CONTRIBUTING.md file, such that someone who wants to verify the installation before using it, knows how to do this.

I have added  a link in the installation guide to point directly to the "Testing" section of [CONTRIBUTING.md](https://github.com/pyvista/pyvista/blob/master/CONTRIBUTING.md#testing). 

PyVista's tests are not deployed with the software but rather are available after cloning the repository so I added some text on that page to let users know that they can run PyVista's comprehensive testing suite after cloning the source and point them to [CONTRIBUTING.md](https://github.com/pyvista/pyvista/blob/master/CONTRIBUTING.md) for further details:

<img width="949" alt="Screen Shot 2019-05-15 at 3 16 36 PM 1" src="https://user-images.githubusercontent.com/22067021/57810193-73fd0300-7724-11e9-992e-f8d80f5c4872.png">

Considering that documentation page is intended to be viewed by a general audience (non-developers), I believe this approach to be welcoming to new users

@Chilipp - do you think our approach here is appropriate?

**Comment by leouieda** [comment]
> Probably not a big issue but I see it's on the checklist - @whedon marks version 0.18.1 in the checklist but PyVista is released officially on 0.20.1

@banesullivan thanks for pointing it out :+1: We'll update this when the review is done. The reviews might lead to changes that should be released before publication.

**Comment by leouieda** [comment]
@nicoguaro @Chilipp thank you so much for the quick reviews! :rocket: As @banesullivan addresses your comments/questions, please make sure you tick the corresponding items from the checklist when you're satisfied.

> The examples work properly, although the colors differ from the documentation.

@banesullivan this might be confusing to other users as well since they'll run the code and get something different from what is "advertised". It might be more transparent to include `pv.set_plot_theme('document')` in the examples. This way, the user also knows that this is possible.

**Comment by banesullivan** [comment]
> this might be confusing to other users as well since they'll run the code and get something different from what is "advertised". It might be more transparent to include pv.set_plot_theme('document') in the examples. This way, the user also knows that this is possible.

That's definitely a fair point - We should make sure users will have the same results "as advertised" when running the examples.

As a starting point, I have added a new example in the *Plotting* section showing off all the different themes in https://github.com/pyvista/pyvista/commit/49444c99d507338fed303341324a2a758f2c4e20

**Comment by Chilipp** [comment]
Thanks for your quick response @banesullivan! Yes, the changes are sufficient for me.

@leouieda: To me this paper is ready for acceptance

**Comment by nicoguaro** [comment]
First, I would like to congratulate the authors for the good work PyVista. I have used VTK directly from Python and this really helps in the process. I have also, used it since it was named vtkInterface and it would say that it is in a mature state now.

Regarding the review:

- I have installed using both ``pip`` and ``conda`` without problems.
- Have checked the examples.
- Have run the tests without problems.
- Built the documentation, and had some (minor) problems when using ``conda``.

The last point is due to some conflicting channels.

Regarding the paper, I have a comment related to the first paragraph and the visualization landscape in Python. The authors do not mention two libraries that I consider that are mature and provide (somewhat) easy to use API, namely:

- [Mayavi](https://arxiv.org/abs/1010.4891v1)
- [yt](https://yt-project.org/)

I would suggest mentioning these in this paragraph.

**Comment by banesullivan** [comment]
Thank you @nicoguaro! That is a fair point about including mentions of other Python visualization software by name. I will make updates to address your concerns and have the paper regenerated. 

Would you please open an issue on the main repository about the conflicting channels or perhaps on the pyvista-feedstock repo (which ever is more relevant): https://github.com/conda-forge/pyvista-feedstock

**Comment by nicoguaro** [comment]
@banesullivan, I did not document what I did. I will repeat it using only ``conda`` and open an issue with the info that I get from there.

**Comment by banesullivan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by nicoguaro** [comment]
@leouieda: To me this paper is ready for acceptance

**Comment by leouieda** [comment]
@nicoguaro thanks for the quick review! :+1:  I see you have a couple of items left in your checklist. Are these intentional or forgotten?

@banesullivan I noticed a few of missing DOIs from the paper:

* Matplotlib: https://doi.org/10.1109/MCSE.2007.55
* Mayavi: https://doi.org/10.1109/MCSE.2011.35
* yt: https://doi.org/10.1088/0067-0049/192/1/9

Could you please add these to the bibtex?

**Comment by banesullivan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by banesullivan** [comment]
@leouieda, thank you for pointing out the missing DOIs. I have updated the `paper.bib` to have DOIs for all references.

One exception being the VTK book - I cannot find a DOI for this but I added an ISBN which `@whedon` doesn't seem to render in the references section of the paper. I filled out our citation for the VTK book per how VTK's website requests (https://vtk.org/about/#citation):

> To cite VTK, please reference the VTK textbook.
>
> Schroeder, Will; Martin, Ken; Lorensen, Bill (2006), The Visualization Toolkit (4th ed.), Kitware, ISBN 978-1-930934-19-1

**Comment by banesullivan** [comment]
@whedon generate pdf from branch joss-patch-figures

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-patch-figures. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by banesullivan** [comment]
@whedon generate pdf from branch joss-patch-figures

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-patch-figures. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by banesullivan** [comment]
@whedon generate pdf from branch joss-patch-figures-2

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-patch-figures-2. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by banesullivan** [comment]
@whedon generate pdf from branch joss-patch-figures-3

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-patch-figures-3. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by banesullivan** [comment]
@whedon generate pdf from branch joss-patch-figures-4

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-patch-figures-4. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by banesullivan** [comment]
Looks like we got it!

@leouieda, can you inspect the last preview of the PDF? The figure looks much better than previously

I'll get a PR going to bring these changes into the master branch

**Comment by banesullivan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by leouieda** [comment]
@banesullivan looks good now :+1: After you merge these changes, please do the following:

* Make a new release of Pyvista with the modifications made during review
* Archive this new version in Zenodo or figshare and post the DOI here

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by banesullivan** [comment]
@leouieda, I have tagged a new release and the CIs are working to publish everything!

Version `0.20.2` is released, being pushed to PyPI, conda-forge, and live on Zenodo.

The Zenodo DOI that always resolves to the latest version is: `10.5281/zenodo.2647611`  (This DOI represents all versions, and will always resolve to the latest one)  Zenodo is still pushing this to DOI.org:

- https://doi.org/10.5281/zenodo.2647611

**Comment by banesullivan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01450/joss.01450/10.21105.joss.01450.pdf)

**Comment by leouieda** [comment]
@whedon set 0.20.2 as version

**Comment by whedon** [comment]
OK. 0.20.2 is the version.

**Comment by leouieda** [comment]
@whedon set 10.5281/zenodo.2647611 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2647611" target="_blank">10.5281/zenodo.2647611</a> is the archive.

**Comment by leouieda** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1109/MCSE.2011.37 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.2514/6.2017-0865 is OK
- 10.1115/1.4041314 is OK
- 10.1115/GT2018-76609 is OK
- 10.1115/GT2018-76800 is OK
- 10.1115/GT2018-76375 is OK
- 10.2514/1.B36849 is OK
- 10.1115/1.4042079 is OK
- 10.1088/0067-0049/192/1/9 is OK
- 10.1109/MCSE.2011.35 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by leouieda** [comment]
OK @banesullivan, given that both reviewers have signed off on this submission, I think we are ready to accept! :confetti_ball:  A minor note: you might want to include ORCIDs of authors (specially yours) in the Zenodo archive.

@nicoguaro and @Chilipp many thanks for the speedy and constructive review! We really appreciate that you donate some of your time to JOSS.

**Comment by leouieda** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1109/MCSE.2011.37 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.2514/6.2017-0865 is OK
- 10.1115/1.4041314 is OK
- 10.1115/GT2018-76609 is OK
- 10.1115/GT2018-76800 is OK
- 10.1115/GT2018-76375 is OK
- 10.2514/1.B36849 is OK
- 10.1115/1.4042079 is OK
- 10.1088/0067-0049/192/1/9 is OK
- 10.1109/MCSE.2011.35 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/696

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/696, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by leouieda** [comment]
@openjournals/joss-eics this submission is ready for publication. I checked the generated PDF and XML and they look good to me.

**Comment by banesullivan** [comment]
> A minor note: you might want to include ORCIDs of authors (specially yours) in the Zenodo archive.

Thanks for pointing this out @leouieda! I didn't realize our ORCIDs were missing (or that we could do this...) I'm struggling to find how/where this goes, do you have any pointers?

**Comment by leouieda** [comment]
> I'm struggling to find how/where this goes, do you have any pointers?

When editing the metadata for authors, you can specify name, affiliation, and ORCID for each one.

**Comment by banesullivan** [comment]
> When editing the metadata for authors, you can specify name, affiliation, and ORCID for each one.

Ah, thanks! I didn't realize I'd have to edit the archive

Relevant to https://github.com/zenodo/zenodo/issues/596

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/702

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/702, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1109/MCSE.2011.37 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.2514/6.2017-0865 is OK
- 10.1115/1.4041314 is OK
- 10.1115/GT2018-76609 is OK
- 10.1115/GT2018-76800 is OK
- 10.1115/GT2018-76375 is OK
- 10.2514/1.B36849 is OK
- 10.1115/1.4042079 is OK
- 10.1088/0067-0049/192/1/9 is OK
- 10.1109/MCSE.2011.35 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
Posted to the Twitters: https://twitter.com/JOSS_TheOJ/status/1130211705118711808

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/703
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01450](https://doi.org/10.21105/joss.01450)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by leouieda** [comment]
@banesullivan congratulations on your publication! :tada: :confetti_ball: This was a nice monday morning surprise :slightly_smiling_face: 

@nicoguaro @Chilipp thank you for taking the time to review this submission in such a short time! :1st_place_medal:

**Comment by banesullivan** [comment]
Thanks @leouieda!!! This is very exciting!

@leouieda, would you please confirm that the following would be a correct BibTex citation for this paper:

```txt
@article{sullivan2019pyvista,
  author = {Sullivan, C. Bane and Kaszynski, Alexander A.},
  journal = {Journal of Open Source Software},
  pages = {4(37), 1450},
  title = {{PyVista: 3D plotting and mesh analysis through a streamlined interface for the Visualization Toolkit (VTK)}},
  year = {2019},
  doi = {10.21105/joss.01450},
}
```

Also, is there a way to autogenerate BibTex citations on the JOSS website? I see the current autogenerated citation of:

> Sullivan et al., (2019). PyVista: 3D plotting and mesh analysis through a streamlined interface for the Visualization Toolkit (VTK). Journal of Open Source Software, 4(37), 1450, https://doi.org/10.21105/joss.01450

**Comment by leouieda** [comment]
Hi @banesullivan I think the bibtex should be:
 
```
@article{Sullivan2019,
  doi = {10.21105/joss.01450},
  url = {https://doi.org/10.21105/joss.01450},
  year = {2019},
  month = may,
  publisher = {The Open Journal},
  volume = {4},
  number = {37},
  pages = {1450},
  author = {C. Sullivan and Alexander Kaszynski},
  title = {{PyVista}: 3D plotting and mesh analysis through a streamlined interface for the Visualization Toolkit ({VTK})},
  journal = {Journal of Open Source Software}
}
```

At least according to https://www.doi2bib.org/bib/10.21105/joss.01450

**Comment by banesullivan** [comment]
Thank you @leouieda!

**Comment by banesullivan** [comment]
@leouieda - we have a slight issue. The publication is showing up under `vtki` still. On the JOSS homepage:

<img width="1620" alt="Screen Shot 2019-05-22 at 7 25 19 PM" src="https://user-images.githubusercontent.com/22067021/58215365-8ef0e980-7cc7-11e9-8cd9-6d5b235d430f.png">


But after clicking on it, all is good:

<img width="1213" alt="Screen Shot 2019-05-22 at 7 25 25 PM" src="https://user-images.githubusercontent.com/22067021/58215371-97492480-7cc7-11e9-93f4-ffe763b64d2d.png">

**Comment by leouieda** [comment]
@banesullivan sorry, I missed this comment. Could you open an issue at https://github.com/openjournals/joss/issues for this? There are other issues related to the front page and it's currently being redesigned. At least the metadata and paper page itself are fine.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01450/status.svg)](https://doi.org/10.21105/joss.01450)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01450">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01450/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01450/status.svg
   :target: https://doi.org/10.21105/joss.01450
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01450/status.svg)](https://doi.org/10.21105/joss.01450)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss

**Comment by banesullivan** [comment]
I noticed recently that the "Software Repository" link on https://joss.theoj.org/papers/10.21105/joss.01450 is incorrect. Can we update this? cc'ing @leouieda and @arfon 

This seems to be an artifact of changing the repository's URL during the pre-review process. Ref https://github.com/openjournals/joss-reviews/issues/1382#issuecomment-491614935

**Comment by arfon** [comment]
OK, I've updated this now.

**Comment by banesullivan** [comment]
Awesome, thank you!!
